public/components/metrics/sidebar/__tests__/sidebar.test.tsx

```diff
@@ -37,16 +41,59 @@ describe('Side Bar Component', () => {
       })
     );
 
-    const wrapper = mount(
-      <Provider store={store}>
-        <Sidebar />
-      </Provider>
-    );
+    let wrapper: ReactWrapper;
+    act(() => {
+      wrapper = mount(
+        <Provider store={store}>
+          <Sidebar />
+        </Provider>
+      );
+    });
 
-    wrapper.update();
+    expect(wrapper.text()).toContain('Available Metrics 0 of 0');
+
+    await act(async () => {
+      console.log('calling resolver');
+      objectListResolver({ observabilityObjectList: [sampleSavedMetric] });
+    });
 
     await waitFor(() => {
-      expect(wrapper).toMatchSnapshot();
+      wrapper.update();
+      expect(wrapper.text()).toContain('Available Metrics 1 of 1');
+    });
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it.skip('renders Side Bar Component for Saved Metric', async () => {
+    let objectListResolver;
+    SavedObjectsActions.getBulk = () =>
+      new Promise((_resolve) => {
+        objectListResolver = _resolve;
+      });
+
+    httpClientMock.get = jest.fn();
+
+    coreRefs.pplService = new PPLService(httpClientMock);
+    coreRefs.pplService.fetch = jest.fn(() => {
+      console.log('mock pplService fetch');
+      return new Promise((resolve) => {
+        resolve({
+          data: { DATA_SOURCES: ['datasource1', 'datasource2'] },
+          then: () => Promise.resolve(),
+        });
+      });
+    });
+
+    const el = document.createElement('div');
+    act(() => {
```

Comment:
try await act here?

Reply:
Can only await an `act(() =>` that is specifically async.   Otherwise, act is speicfically a sync call and acts that way in all respects.
The first act() is setting up render, which is not an async call.
Evidence : The initial page render is "all blank" b/c dependency data is entirely missing.  The first visible page render is "Available Metrics 0 of 0", which is non-blank, so ensures first render cycle has executed.